README.md

```diff
@@ -280,5 +221,123 @@ it is a date in our "b" relation.
 
 ```
 
+## compare_all_columns ([source](macros/compare_all_columns.sql))
+This macro is designed to be added to a dbt test suite as a custom test. A 
+`compare_all_columns` test monitors changes data values when code is changed 
+as part of a PR or during development. It sets up a test that will fail 
+if any column values do not match. 
+
+Users can configure what exactly constitutes a value match or failure. If 
+there is a test failure, results can be inspected in the warehouse. The primary key 
+and the column name can be included in the test output that gets written to the warehouse. 
+This enables the user to join test results to relevant tables in your dev or prod schema to investigate the error.
+
+### Usage:
+
+_Note: this test should only be used on (and will only work on) models that have a primary key that is reliably `unique` and `not_null`. [Generic dbt tests](https://docs.getdbt.com/docs/building-a-dbt-project/tests#generic-tests) should be used to ensure the model being tested meets the requirements of `unique` and `not_null`._
+
+To create a test for the `stg_customers` model, create a custom test 
+in the `tests` subdirectory of your dbt project that looks like this:
+
+```
+{{ 
+  audit_helper.compare_all_columns(
+    a_relation=ref('stg_customers'), -- in a test, this ref will compile as your dev or PR schema.
+    b_relation=api.Relation.create(database='dbt_db', schema='analytics_prod', identifier='stg_customers'), -- you can explicitly write a relation to select your production schema, or any other db/schema/table you'd like to use for comparison testing.
```

Comment:
I'd probably add a note here saying you can also hard code a table name - for interactive queries where someone is writing one-off code, it's not unreasonable to hardcode a table string for expediency's sake. If you're building it into a CI cycle, then yes please make a proper source/relation/etc